### DISCUSSION TITLE
I'm curious if the Data(p) part is not encrypted in FastDDS with SROS2 applied.

### DISCUSSION THREAD

**Discussion by dhkstnrh** [eProsima/Fast-DDS#3945] [Q&A]
Hello!

I'm a student studying ROS2.

After implementing SROS2 in FastDDS and inspecting the packets with WireShark, I noticed that the Data(p) part sent to the Global Data Space was not encrypted. Upon reviewing the official DDS-Security documentation, it mentions that the RTPS Header is not encrypted. What is the reason for not encrypting this RTPS Header? I'm also curious about why the information in the Data(p) part is not encrypted!

![image](https://github.com/eProsima/Fast-DDS/assets/55154297/46d32f71-c054-4970-b5fe-f59acdd73f1d)

**Comment by officalLEE** [comment]
Yes sir! me too i wanna know about it!!

**Comment by JLBuenoLopez** [comment]
@dhkstnrh 

Please, refer to the [DDS Security specification](https://www.omg.org/spec/DDS-SECURITY/1.1/About-DDS-SECURITY) section 7.4.1. That section includes the rationale of why the ParticipantBuiltinTopicData is not protected.

**Reply by dhkstnrh** [reply]
Thank you for your answer!
